SalesforceFoundation/EDA
# [EDA]: Prevent Deletion of Course Offering if There are Associated Child Records

# Critical Changes

# Changes
## Updates to Ensure Data Integrity
We've added functionality to prevent the deletion of Course Offering records that have related Term Grade, Course Connection, or Course Offering Schedule records. This change is for internal-use only and is disabled at this time. You won't see any difference in functionality.
**Note:** We strongly recommend that EDA Admins don't manage Hierarchy Custom Settings directly and, instead, use the EDA Settings UI.

# Issues Closed
Closes #1047
# New Metadata
COFF_CannotDelete_TDTM
COFF_CannotDelete_TEST

# Deleted Metadata

# Testing Notes
* Scenario 1: “Prevent Course Offering Deletion” checkbox is enabled and Course Offering record has related Term Grade, Course Connection, or Course Offering Schedule records. 
    * Check “Prevent Course Offering Deletion” checkbox in Hierarchy Settings. 
    * Create an Academic Program (or a different Account record Type). 
    * Create Term and Course records. 
    * Create a Course Offering record. 
    * Add a Term Grade, Course Connection, Course Offering Schedule records or all three to the Course Offering record. 
    * Delete the Course Offering record. 
    * You should receive a pop up error that you cannot delete this Course Offering. Seehttps://quip.com/x5xKAGcHko2L#fIEACAMe5GG for the error message. 
* Scenario 2: “Prevent Course Offering Deletion” checkbox is enabled and Course Offering record does not have related Term Grade, Course Connection, or Course Offering Schedule records. 
    * Check “Prevent Course Offering Deletion” checkbox in Hierarchy Settings. 
    * Create an Academic Program (or a different Account record Type). 
    * Create Term and Course records. 
    * Create a Course Offering record. 
    * Delete the Course Offering record. 
    * You should be able to delete Course Offering record. 
* Scenario 3: “Prevent Course Offering Deletion” checkbox is disabled and Course Offering record has related Term Grade, Course Connection, or Course Offering Schedule records. 
    * Check “Prevent Course Connection Deletion” checkbox in Hierarchy Settings. 
    * Create an Academic Program (or a different Account record Type). 
    * Create Term and Course records. 
    * Create a Course Offering record. 
    * Add a Term Grade, Course Connection, Course Offering Schedule records or all three to the Course Offering record. 
    * Delete the Course Offering record. 
    * You should be able to delete Course Offering record. 
* Scenario 4: “Prevent Course Offering Deletion” checkbox is disabled and Course Offering record does not have related Term Grade, Course Connection, or Course Offering Schedule records. 
    * Check “Prevent Course Connection Deletion” checkbox in Hierarchy Settings. 
    * Create an Academic Program (or a different Account record Type). 
    * Create Term and Course records. 
    * Create a Course Offering record. 
    * Delete the Course Offering record. 
    * You should be able to delete Course Offering record.
